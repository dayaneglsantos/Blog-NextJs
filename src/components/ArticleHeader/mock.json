{
  "publishedAt": "2023-12-20T17:23:29.387Z",
  "title": "TÃ­tulo da postagem",
  "excerpt": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Vero ratione quisquam alias doloribus veniam dolore. Ea temporibus illum numquam ab quo.",
  "image": "/image.com",
  "author": {
    "slug": "dayane-santos",
    "displayName": "Dayane Santos"
  },
  "categories": [
    {
      "slug": "react",
      "displayName": "React"
    },
    {
      "slug": "js",
      "displayName": "Javascript"
    }
  ]
}
