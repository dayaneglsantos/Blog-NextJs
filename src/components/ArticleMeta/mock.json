{
  "publishedAt": "2023-12-20T17:23:29.387Z",
  "darkTheme": false,
  "author": {
    "slug": "dayane-santos",
    "displayName": "Dayane Santos"
  },
  "categories": [
    {
      "slug": "react",
      "displayName": "React"
    },
    {
      "slug": "js",
      "displayName": "Javascript"
    }
  ]
}
