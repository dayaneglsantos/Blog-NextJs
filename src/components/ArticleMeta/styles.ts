'use client'
import styled from 'styled-components'

export const ArticleMetaContainer = styled.div`
  font-style: italic;
  margin-top: 12px;

  p {
    color: ${({ theme }) => theme.colors.secondaryFont};
  }
  .after::after {
    content: ', ';
  }

  .after:last-child::after {
    content: '';
  }
`
