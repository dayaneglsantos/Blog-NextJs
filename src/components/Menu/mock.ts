import { MenuType } from '.'

export default {
  links: [
    {
      link: '/',
      text: 'Home',
      newTab: true
    },
    {
      link: '/about',
      text: 'About',
      newTab: true
    },
    {
      link: '/posts',
      text: 'Posts',
      newTab: true
    },
    {
      link: '/fallow-us',
      text: 'Fallow us',
      newTab: true
    },
    {
      link: '/contact',
      text: 'Contact',
      newTab: true
    }
  ]
} as MenuType
