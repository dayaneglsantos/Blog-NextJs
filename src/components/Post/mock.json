{
  "title": "Post 2",
  "slug": "post-2",
  "alternativeText": "Foto 2",
  "image": "https://res.cloudinary.com/dbmuc0tzr/image/upload/v1703092587/img5_565f978262.jpg",
  "excerpt": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Vero ratione quisquam alias doloribus veniam dolore",
  "textContent": "<p>Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, commodo vitae, ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. Donec non enim in turpis pulvinar facilisis. Ut felis. Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus</p><ul><li>Morbi in sem quis dui placerat ornare. Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu. Cras consequat.</li><li>Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus.</li><li>Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique cursus. Nam nulla quam, gravida non, commodo a, sodales sit amet, nisi.</li><li>Pellentesque fermentum dolor. Aliquam quam lectus, facilisis auctor, ultrices ut, elementum vulputate, nunc.</li></ul>",
  "allowComments": true,
  "publishedAt": "2023-12-20T17:23:29.387Z",
  "cover": "/cover.com.br/img",
  "altText": "Alt text do cover",
  "id": "1",

  "categories": [
    {
      "displayName": "React",
      "slug": "react"
    },
    {
      "displayName": "Front-end",
      "slug": "fron-end"
    }
  ],
  "tags": [
    {
      "displayName": "Tag 1",
      "slug": "tag-1"
    },
    {
      "displayName": "Tag 2",
      "slug": "tag-2"
    }
  ],
  "author": {
    "displayName": "Dayane Santos",
    "slug": "dayane-santos"
  }
}
