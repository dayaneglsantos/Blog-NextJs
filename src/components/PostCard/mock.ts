import { data } from '../../Api/data.json'

export default data.posts.data[0].attributes
