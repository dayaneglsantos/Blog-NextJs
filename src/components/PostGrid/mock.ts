import { PostGridType } from '.'

export default {
  posts: [
    {
      id: '1',
      slug: 'slug-1',
      excerpt:
        "There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text.",
      title: 'Neque porro quisquam est qui dolorem',
      cover:
        'https://images.unsplash.com/photo-1682687982502-1529b3b33f85?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    },
    {
      id: '2',
      slug: 'slug-2',
      excerpt:
        'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using.',
      title: 'There are many variations of passages of Lorem',
      cover:
        'https://images.unsplash.com/photo-1703877048317-c74856d65c42?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    },
    {
      id: '3',
      slug: 'slug-3',
      excerpt:
        'The standard chunk of Lorem Ipsum used since the 1500s is reproduced below for those interested. Sections 1.10.32 and 1.10.33 from de Finibus Bonorum et Malorum by Cicero are also reproduced their exact original form, accompanied by English versions from the 1914 translation by H. Rackham.',
      title: 'Sed ut perspiciatis unde omnis iste natus error',
      cover:
        'https://images.unsplash.com/photo-1703701008461-ce64767019d3?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    },
    {
      id: '1',
      slug: 'slug-1',
      excerpt:
        "There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text.",
      title: 'Neque porro quisquam est qui dolorem',
      cover:
        'https://images.unsplash.com/photo-1682687982502-1529b3b33f85?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    },
    {
      id: '2',
      slug: 'slug-2',
      excerpt:
        'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using.',
      title: 'There are many variations of passages of Lorem',
      cover:
        'https://images.unsplash.com/photo-1703877048317-c74856d65c42?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    },
    {
      id: '3',
      slug: 'slug-3',
      excerpt:
        'The standard chunk of Lorem Ipsum used since the 1500s is reproduced below for those interested. Sections 1.10.32 and 1.10.33 from de Finibus Bonorum et Malorum by Cicero are also reproduced their exact original form, accompanied by English versions from the 1914 translation by H. Rackham.',
      title: 'Sed ut perspiciatis unde omnis iste natus error',
      cover:
        'https://images.unsplash.com/photo-1703701008461-ce64767019d3?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
    }
  ],
  disabled: false,
  noMorePosts: false,
  onclick: jest.fn()
} as PostGridType
