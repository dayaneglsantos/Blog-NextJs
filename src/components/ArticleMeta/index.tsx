import Link from 'next/link'
import { formatDate } from '../../Utils/format-date'
import { Author } from '../../shared-types/author'
import { Category } from '../../shared-types/category'
import { ArticleMetaContainer } from './styles'

export type ArticleMetaType = {
  publishedAt: string
  author: Author
  categories?: Category[]
}

const ArticleMeta = ({
  author,
  publishedAt,
  categories = []
}: ArticleMetaType) => {
  return (
    <ArticleMetaContainer>
      <p>
        Por <Link href={`/author/${author.slug}`}>{author.displayName} </Link>
        em <span>{formatDate(publishedAt)}</span> |{' '}
        {categories.length > 0 && (
          <span>
            {categories.map((category) => {
              return (
                <Link
                  href={`/category/${category.slug}`}
                  key={category.slug}
                  className="after"
                >
                  {category.displayName}
                </Link>
              )
            })}
          </span>
        )}
      </p>
    </ArticleMetaContainer>
  )
}

export default ArticleMeta
