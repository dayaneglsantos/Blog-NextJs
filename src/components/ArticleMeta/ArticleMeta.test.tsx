import { screen } from '@testing-library/react'
import ArticleMeta from '.'
import mock from './mock.json'
import { renderTheme } from '../../Utils/render-theme'

describe('<ArticleMeta/>', () => {
  it('should render ArticleMeta', () => {
    renderTheme(<ArticleMeta {...mock} />)
    const author = screen.getByRole('link', { name: 'Dayane Santos' })
    expect(author).toBeInTheDocument()

    const category = screen.getByRole('link', { name: 'React' })
    expect(category).toHaveAttribute('href', '/category/react')
  })
  it('should render ArticleMeta without categories', () => {
    renderTheme(<ArticleMeta {...mock} categories={undefined} />)

    const category = screen.queryByRole('link', { name: 'React' })
    expect(category).not.toBeInTheDocument()
  })
  it('should match snapshot', () => {
    const container = renderTheme(<ArticleMeta {...mock} />)
    expect(container).toMatchSnapshot()
  })
})
