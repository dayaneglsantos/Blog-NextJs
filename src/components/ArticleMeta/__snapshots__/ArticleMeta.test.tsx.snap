// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleMeta/> should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  font-style: italic;
  margin-top: 12px;
}

.c0 p {
  color: #495E57;
}

.c0 .after::after {
  content: ', ';
}

.c0 .after:last-child::after {
  content: '';
}

<body>
    <div>
      <div
        class="c0"
      >
        <p>
          Por 
          <a
            href="/author/dayane-santos"
          >
            Dayane Santos
             
          </a>
          em 
          <span>
            20/12/2023, 14:23
          </span>
           |
           
          <span>
            <a
              class="after"
              href="/category/react"
            >
              React
            </a>
            <a
              class="after"
              href="/category/js"
            >
              Javascript
            </a>
          </span>
        </p>
      </div>
    </div>
  </body>,
  "container": .c0 {
  font-style: italic;
  margin-top: 12px;
}

.c0 p {
  color: #495E57;
}

.c0 .after::after {
  content: ', ';
}

.c0 .after:last-child::after {
  content: '';
}

<div>
    <div
      class="c0"
    >
      <p>
        Por 
        <a
          href="/author/dayane-santos"
        >
          Dayane Santos
           
        </a>
        em 
        <span>
          20/12/2023, 14:23
        </span>
         |
         
        <span>
          <a
            class="after"
            href="/category/react"
          >
            React
          </a>
          <a
            class="after"
            href="/category/js"
          >
            Javascript
          </a>
        </span>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
